packages/client/src/Loc.ts

```diff
@@ -2070,16 +2073,20 @@ export class ClosedLoc extends LocRequestState {
 
 export class ClosedIdentityLoc extends ClosedLoc {
 
-    get secrets(): Secret[] {
-        return Object.values(this._secrets);
-    }
-
-    addSecret(secret: Secret) {
-        this._secrets[secret.name] = secret;
+    async addSecret(secret: Secret): Promise<ClosedIdentityLoc> {
+        this.locSharedState.client.addSecret({
+            locId: this.locId,
+            secret,
+        });
+        return await this.refresh() as ClosedIdentityLoc;
     }
```

Comment:
Should `await` backend operation completion before refreshing.

```suggestion
    async addSecret(secret: Secret): Promise<ClosedIdentityLoc> {
        await this.locSharedState.client.addSecret({
            locId: this.locId,
            secret,
        });
        return await this.refresh() as ClosedIdentityLoc;
    }
```